src/display/canvas.js

```diff
@@ -1121,6 +1125,8 @@ class CanvasGraphics {
     }
     this._cachedScaleForStroking = null;
     this._cachedGetSinglePixelWidth = null;
+    this._cachedBitmaps = [];
+    this._cachedBitmapsMap = new WeakMap();
```

Comment:
Do we need *two* different, but closely related, caches here since it feels like it's adding some overall complexity to the code?

Reply:
Yep it's because I chose a WeakMap and I added this array in order to be able to clean up the data at the end.
But likely a WeakMap is useless here and a Map should be enough.